qiskit-superstaq/qiskit_superstaq/superstaq_backend.py

```diff
@@ -136,3 +206,234 @@ def run(
+        )
+
+    def aqt_compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        num_equivalent_circuits: Optional[int] = None,
+        random_seed: Optional[int] = None,
+        atol: Optional[float] = None,
+        **kwargs: Any,
+    ) -> qss.compiler_output.CompilerOutput:
+        """Compiles and optimizes the given circuit(s) for the Advanced Quantum Testbed (AQT) at
+        Lawrence Berkeley National Laboratory. Also allows using Equivalent Circuit Averaging (ECA).
+
+        See arxiv.org/pdf/2111.04572.pdf for a description of ECA.
+
+        Args:
+            circuits: The qiskit QuantumCircuit(s) to compile.
+            num_equivalent_circuits: Number of logically equivalent random circuits to generate for
+                each input circuit.
+            random_seed: Optional seed for circuit randomizer.
+            atol: Tolerance to use for approximate gate synthesis (currently just for qutrit gates).
+            kwargs: Other desired compile options.
+
+        Returns:
+            Object whose .circuit(s) attribute contains the optimized circuits(s). Alternatively for
+            ECA, an Object whose .circuits attribute is a list (or list of lists) of logically
+            equivalent circuits If qtrl is installed, the object's .seq attribute is a qtrl Sequence
+            object containing pulse sequences for each compiled circuit, and its .pulse_list(s)
+            attribute contains the corresponding list(s) of cycles.
+
+        Raises:
+            ValueError: If `target` is not a valid AQT target.
+        """
+        if not self.name().startswith("aqt_"):
+            raise ValueError(f"{self.name()} is not a valid AQT target.")
+
+        options: Dict[str, Any] = {**kwargs}
+        if num_equivalent_circuits is not None:
+            options["num_equivalent_cirucits"] = num_equivalent_circuits
+        if random_seed is not None:
+            options["random_seed"] = random_seed
+        if atol is not None:
+            options["atol"] = atol
+
+        serialized_circuits = qss.serialization.serialize_circuits(circuits)
+        request_json = {
+            "qiskit_circuits": serialized_circuits,
+            "target": self.name(),
+            "options": json.dumps(options),
+        }
+        metadata_of_circuits = _get_metadata_of_circuits(circuits)
+        circuits_is_list = not isinstance(circuits, qiskit.QuantumCircuit)
+        json_dict = self._provider._client.aqt_compile(request_json)
+        return qss.compiler_output.read_json_aqt(
+            json_dict, metadata_of_circuits, circuits_is_list, num_equivalent_circuits
+        )
+
+    def ibmq_compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        **kwargs: Any,
+    ) -> qss.compiler_output.CompilerOutput:
+        """Gets result of IBMQ compilation request.
```

Comment:
should be updated to something like
```suggestion
        """Compiles and optimizes the given circuit(s) for IBMQ devices.
```
(ditto for cq_compile)

Reply:
(Done for cq_compile.)